### PR TITLE
Show JIT channel fee in wallet history

### LINKED ISSUE #982: Show JIT Channel fee as fee in the wallet history
Currently the JIT Channel fee is only shown as payment with the payment hash as description. We should align that with the order matching fees. e.g.

- [ ] Dedicated icon (or the same as for order matching)
- [ ] Description that the fee was for opening an inbound channel.

Screenshot 2023-07-23 at 06 34 06

### PR DESCRIPTION
**Pull request by Restioson**
image

Fixes #982

### CODE REVIEW

**Comment by holzeis** [comment]
@Restioson That was not the intention of that ticket 🙈. I suggested a similar display as for the matching fees 😅 

Screenshot 2023-08-16 at 13 40 49

This change goes more into the direction of showing transaction details. https://github.com/get10101/10101/issues/872 Did you maybe linked the wrong ticket?

**Comment by Restioson** [comment]
I didn't link the wrong ticket - check main since #1054 has been merged. This PR extends #1054 by separating JIT opening fee from regular payments

File: crates/orderbook-commons/src/lib.rs
Comment (luckysori) [inline crates/orderbook-commons/src/lib.rs:24]: 📣 I appreciate that it's probably nicer to remove this dash, but I think this is a breaking change. Probably worth avoiding that headache.

As a side note, I am pretty sure the prefix idea that I introduced is not very good and there were better alternatives 🙈

File: crates/orderbook-commons/src/lib.rs
Comment (luckysori) [inline crates/orderbook-commons/src/lib.rs:28]: 🤔 Hmmm, this is weird because the JIT channel is orthogonal to the orderbook. Maybe put this in `ln-dlc-node` for now?

**Review by luckysori — COMMENTED** [review]
LGTM! I left a couple of remarks that are worth considering.

File: crates/orderbook-commons/src/lib.rs
Comment (Restioson) [reply crates/orderbook-commons/src/lib.rs:24]: > 📣 I appreciate that it's probably nicer to remove this dash, but I think this is a breaking change. Probably worth avoiding that headache.

Yea, most definitely :sweat_smile: 

Maybe for now I should add a legacy match.

The reason I added this was because there isn't really a way to match on multiple prefixes cleanly, whereas if you define the syntax for a structured payment invoice description as 

```
{tag} {additional_data}
```

then you can just `split_once` by space and match on the tag to get what kind of invoice it is.

However, it's worth noting that both this implementation _and_ the previous implementation could lead to some strange results. 

For instance, what would happen if an invoice were paid by a mischievous (or even malicious) user with the description `taker-fee-NotAnOrderId?!/.sidfj`? At best this would display wrong, worse it could crash the app, and at worst it could introduce a security vulnerability.

It's probably worth workshopping how this should work :sweat_smile:

File: crates/orderbook-commons/src/lib.rs
Comment (Restioson) [reply crates/orderbook-commons/src/lib.rs:28]: True, can do!

**Comment by Restioson** [comment]
I've pushed a commit to revert, and happy to squash up once we've resolved the discussion. Please don't merge until then

**Comment by holzeis** [comment]
> I didn't link the wrong ticket - check main since #1054 has been merged :slightly_smiling_face: This PR extends #1054 by separating JIT opening fee from regular payments

Hmm.. on main it's now simply shown as an outgoing payment instead of showing directly in the title that it is a JIT channel open fee payment. It seems inconsistent that we show a matching fee in the title but not for the JIT channel opening fee.

**Comment by Restioson** [comment]
I might not be understanding you correctly here so bear with me as I repeat what I understand from what you said :sweat_smile: 

> Hmm.. on main it's now simply shown as an outgoing payment instead of showing directly in the title that it is a JIT channel open fee payment

Agree, that's what this PR is changing.

> It seems inconsistent that we show a matching fee in the title but not for the JIT channel opening fee

On this branch, it does show as channel open fee. Let me show another screenshot in case it wasn't clear:

image

**Comment by holzeis** [comment]
> I might not be understanding you correctly here so bear with me as I repeat what I understand from what you said :sweat_smile: 
> 
> 
> 
> > Hmm.. on main it's now simply shown as an outgoing payment instead of showing directly in the title that it is a JIT channel open fee payment
> 
> 
> 
> Agree, that's what this PR is changing.
> 
> 
> 
> > It seems inconsistent that we show a matching fee in the title but not for the JIT channel opening fee
> 
> 
> 
> On this branch, it does show as channel open fee. Let me show another screenshot in case it wasn't clear:
> 
> 
> 
> image
> 
> 

Oh that's great! I just saw the other screenshot and thought this was the only change 😅

Well done 👍 looks nice. Sorry for the confusion 😅

**Comment by Restioson** [comment]
No problem!

So I guess it's just the query about prefixes vs splitting to resolve? Maybe that should go in a follow-up and just stick to prefixes for now, though. In that case I can rebase this and squash it up before merge

**Review by holzeis — APPROVED** [review]
LGTM 👍

**Pull request merged**
